### PR TITLE
Updates to sync streaming inputs/outputs.

### PR DESCRIPTION
**Pull request by shorea**
Still missing the generator changes to add overloads for file and input stream but wanted to get some feedback on what I've got so far.

### CODE REVIEW

**Comment by shorea** [comment]
Updating documentation and fixing build.

**Comment by shorea** [comment]
Moar build fixes...

File: services/s3/src/it/java/utils/BucketNamingStrategy.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import static software.amazon.awssdk.utils.StringUtils.lowerCase;
+
+/**
+ * Naming strategy for S3 buckets that uses the test class name and the user name to come up with a unique but reproducible
+ * bucket name.
+ */
+public class BucketNamingStrategy {
+
+    public String getBucketName(Class<?> testClass) {
+        return String.format("%s-%s", lowerCase(testClass.getSimpleName()), lowerCase(System.getProperty("user.name")));
```
Comment (varunnvs92) [inline services/s3/src/it/java/utils/BucketNamingStrategy.java]: Nice.
Is there chance of System.getProperty("user.name") returning null?

File: codegen/src/main/java/software/amazon/awssdk/codegen/docs/Factory.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.util.function.BiFunction;
+import software.amazon.awssdk.codegen.model.intermediate.IntermediateModel;
+import software.amazon.awssdk.codegen.model.intermediate.OperationModel;
+
+/**
+ * Factory interface to reduce the verbosity of using BiFunction with type args everywhere.
+ */
+@FunctionalInterface
+interface Factory extends BiFunction<IntermediateModel, OperationModel, OperationDocProvider> {
```
Comment (shorea) [inline codegen/src/main/java/software/amazon/awssdk/codegen/docs/Factory.java]: Sure thing, It was originally an inner interface but I must have factored it out at some point.

**Pull request merged**